org.eclipse.emfcloud.validation/src/main/java/org/eclipse/emfcloud/validation/ValidationFramework.java

```diff
@@ -0,0 +1,264 @@
+import org.eclipse.emfcloud.modelserver.client.SubscriptionListener;
+import org.jetbrains.annotations.NotNull;
+
+public class ValidationFramework {
+
+    private String defaultURL = "http://localhost:8081/api/v1/";
+
+    private String modelUri;
+
+    private ModelServerClient modelServerClient;
+
+    public List<ValidationResult> recentValidationResult = new ArrayList<>();
+
+    public ValidationResultChangeListener changeListener;
+
+    public Map<Integer, Map<Integer, EMFFacetConstraints>> inputValidationMap = new HashMap<>();
+
+    private List<ValidationFilter> validationFilterList = new ArrayList<>();
+
+    public ValidationFramework(String modelUri, ValidationResultChangeListener changeListener) throws MalformedURLException {
+        this.modelUri = modelUri;
+        this.modelServerClient = new ModelServerClient(defaultURL);
+        this.changeListener = changeListener;
+    }
+
+    public ValidationFramework(String modelUri, ModelServerClient modelServerClient, ValidationResultChangeListener changeListener) {
+        this.modelUri = modelUri;
+        this.modelServerClient = modelServerClient;
+        this.changeListener = changeListener;
+    }
+
+    public CompletableFuture<Void> validate() throws IOException, InterruptedException, ExecutionException {
+        return this.modelServerClient.validate(modelUri).thenAccept(s -> {
+            try {
+                readData(modelUri, s.body());
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        });
+    }
+
+    public CompletableFuture<Void> getConstraintList() {
+        return this.modelServerClient.getConstraints(modelUri).thenAccept(s -> {
+            try {
+                readConstraintList(s.body());
+            } catch (IOException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        });
+    }
+
+    public void subscribeToValidation() {
+        this.modelServerClient.validationSubscribe(modelUri, new SubscriptionListener() {
+            @Override
+            public void onOpen(Response<String> response) {
+                System.out.println("Connected: " + response.getMessage());
+                try {
+                    validate();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                } catch (InterruptedException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                } catch (ExecutionException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onClosing(int code, @NotNull String reason) {
+                System.out.println("Closing: Code " + code);
```

Comment:
do we need this?

Reply:
Do you mean the `System.out.println()` or the SubscriptionListener()?

Reply:
all the sysouts